src/nvidia.cc

```diff
@@ -827,27 +888,37 @@ void print_nvidia_value(struct text_object *obj, char *p, int p_max_size)
 	struct nvidia_s *nvs = static_cast<nvidia_s *>(obj->data.opaque);
 	int value, temp1, temp2;
 	char* str;
+	
+    Display *dpy = nvdisplay ? nvdisplay : display;
+    
+    //num_GPU calculated only once based on the physical target
+	static int num_GPU = get_nvidia_target_count(dpy, TARGET_GPU) - 1;
```

Comment:
Shouldn't this also be done in `get_nvidia_barval`? Also why the -1? Is that related to line 905? If so would it be better to move that logic there?

Also, if the same error checking and gpu id reducing needs to be done in both print_nvidia_value and get_nvidia_barval, should it be moved into a separate inline function or macro?